src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/vm/AzureVMsProvider.java

```diff
@@ -0,0 +1,295 @@
+                                                                                                             image,
+                                                                                                             creatableNetworkInterface);
+                                                                            })
+                                                                            .collect(Collectors.toList());
+
+        // Create all VMs in parallel and collect IDs
+        return azureService.virtualMachines()
+                           .create(creatableVirtualMachines)
+                           .values()
+                           .stream()
+                           .map(vm -> instance.withTag(vm.name()).withId(vm.vmId()).withNumber(SINGLE_INSTANCE_NUMBER))
+                           .collect(Collectors.toSet());
+    }
+
+    private Creatable<VirtualMachine> prepareVitualMachine(Instance instance, Azure azureService,
+            ResourceGroup resourceGroup, Region region, String instanceTag, VirtualMachineCustomImage image,
+            Creatable<NetworkInterface> creatableNetworkInterface) {
+
+        // Configure the VM depending on the OS type
+        VirtualMachine.DefinitionStages.WithFromImageCreateOptionsManaged creatableVirtualMachineWithImage;
+        OperatingSystemTypes operatingSystemType = image.osDiskImage().osType();
+        if (operatingSystemType.equals(OperatingSystemTypes.LINUX)) {
+            creatableVirtualMachineWithImage = configureLinuxVirtualMachine(azureService,
+                                                                            instanceTag,
+                                                                            region,
+                                                                            resourceGroup,
+                                                                            instance.getCredentials(),
+                                                                            image,
+                                                                            creatableNetworkInterface);
+        } else if (operatingSystemType.equals(OperatingSystemTypes.WINDOWS)) {
+            creatableVirtualMachineWithImage = configureWindowsVirtualMachine(azureService,
+                                                                              instanceTag,
+                                                                              region,
+                                                                              resourceGroup,
+                                                                              instance.getCredentials(),
+                                                                              image,
+                                                                              creatableNetworkInterface);
+        } else {
+            throw new RuntimeException("ERROR Operating System of type '" + operatingSystemType.toString() +
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)